pkg/harvester/edit/migration.harvesterhci.io.virtualmachineimport.vue

```diff
@@ -0,0 +1,474 @@
+      } else if (this.sourceProviderType === 'ova') {
+        list = this.ovaSources;
+      }
+
+      return list.map((s) => {
+        // Fallback for API version/kind if missing on the object
+        let kind = s.kind;
+
+        if (!kind) {
+          if (this.sourceProviderType === 'vmware') kind = VMWARE_KIND;
+          else if (this.sourceProviderType === 'openstack') kind = OPENSTACK_KIND;
+          else if (this.sourceProviderType === 'ova') kind = OVA_KIND;
+        }
+
+        const apiVersion = s.apiVersion || `${ MIGRATION_GROUP }/v1beta1`;
+
+        return {
+          label: s.metadata.name,
+          value: `${ kind }/${ s.metadata.namespace }/${ s.metadata.name }`,
+          // We attach the raw metadata so we can easily populate the spec later without re-finding the object
+          raw:   {
+            kind,
+            apiVersion,
+            name:      s.metadata.name,
+            namespace: s.metadata.namespace
+          }
+        };
+      });
+    },
+
+    // Perform various form validations before allowing to submit
+    isFormValid() {
+      // Check VM Name is valid
+      const nameError = this.fvNameRule(this.value.spec.virtualMachineName);
+
+      if (nameError) return false;
+
+      // Check mandatory fields in Basics
+      if (!this.value.spec.virtualMachineName) return false;
+      if (!this.selectedSourceKey) return false;
+
+      // Check Network Mappings
+      // If any row is missing source or destination, the form is invalid.
+      const networks = this.value.spec.networkMapping || [];
+      const hasInvalidRow = networks.some((row) => !row.sourceNetwork || !row.destinationNetwork);
+
+      if (hasInvalidRow) return false;
+
+      return true;
+    },
+
+    // Filter out internal storage classes
+    // to prevent selecting a class that might cause the import to fail
+    storageClassOptions() {
+      return this.allStorageClasses
+        .filter((sc) => {
+          const isInternal = sc.parameters?.['harvesterhci.io/isInternalStorageClass'] === 'true';
+
+          return !isInternal;
+        })
+        .map((sc) => ({
+          label: sc.nameDisplay,
+          value: sc.id
+        }));
+    },
+
+    networkOptions() {
+      return this.allNetworks.map((n) => ({
+        label: n.nameDisplay || n.metadata.name,
+        value: n.id
+      }));
+    }
+  },
+
+  methods: {
+    // Clear the selected cluster if the user switches providers (e.g. VMware -> OpenStack)
+    // Prevents submitting a VMware cluster name while the kind is OpenStack
+    onProviderTypeChange(newType) {
+      this.selectedSourceKey = null;
+      this.value.spec.sourceCluster = {};
+    },
+
+    // Update the sourceCluster object based on the single dropdown selection
+    updateSource(key) {
+      this.selectedSourceKey = key;
+      const selectedOption = this.sourceOptions.find((o) => o.value === key);
+
+      if (selectedOption) {
+        this.value.spec.sourceCluster = {
```

Comment:
[nit] can be shortened to 
```suggestion
const { kind, apiVersion, name, namespace} = selectedOption.raw;
        this.value.spec.sourceCluster = {
          kind, 
          apiVersion,
          name,
          namespace
        }
```